web/src/queries/storage/config-model.ts

```diff
@@ -252,6 +278,23 @@ export function useBoot(): BootHook {
   };
 }
 
+export type PartitionHook = {
+  onDelete: () => void;
+};
+
+// driveName, like "/dev/sda"
+// mountPath, like "/" or "swap"
+export function usePartition(driveName: string, mountPath: string): PartitionHook | undefined {
+  const model = useConfigModel();
+  const { mutate } = useConfigModelMutation();
+
+  if (findPartition(model, driveName, mountPath) === undefined) return;
+
+  return {
+    onDelete: () => mutate(deletePartition(model, driveName, mountPath)),
```

Comment:
NP: I'd call the mutation just _delete_ and keep the _onDelete_ name for naming the event/callback on components.

Reply:
I tried that but had a problem with that, apparently `delete` is a reserved word somewhere. I will retry with quoting :)